League team player categories

VasilyShelkov opened the pull request:
reference #44 
close #1 
close #55 

@Winwardo Ready for comments and merging

VasilyShelkov committed:
Initial attempt at adding league

VasilyShelkov committed:
Merged branch master into league-team-player-categories

VasilyShelkov committed:
Completed LeagueCategory

VasilyShelkov committed:
Got to calling the football teams enpoint but not working

VasilyShelkov committed:
Got TeamsByLeague working, just got to fix the accordian and search

VasilyShelkov committed:
Only Searching 2015 + fetching the league teams of open tab

VasilyShelkov committed:
Got the search in Teams properly working

VasilyShelkov committed:
Added a pointed cursor for selecting a team

VasilyShelkov committed:
Slight changes to the year tabs to make API calls only once

VasilyShelkov committed:
Throttling the search term changes

VasilyShelkov committed:
General Refactoring

VasilyShelkov committed:
Added loader if retrieving the leagues teams

VasilyShelkov committed:
Fetch Teams before doing the other years

VasilyShelkov committed:
changed variable name

VasilyShelkov committed:
Added football players endpoint

VasilyShelkov committed:
Added tests and football team players reducer cases

VasilyShelkov committed:
Added Players Category working

VasilyShelkov committed:
Addded styling and buttons for properly selecting teams

VasilyShelkov committed:
added RemoveSelectedFootballTeam + small tweaks

- removes the selected football when changing tabs
- when searching through the teams, shows a count of how many there against how many are filtered

VasilyShelkov committed:
Added animation to indicate what the numbers on the tabs mean

VasilyShelkov committed:
Added comments for the league count animation

VasilyShelkov committed:
Addded tab year animation

VasilyShelkov committed:
Correctly formatted the filters

VasilyShelkov committed:
npm run lint

VasilyShelkov committed:
Merged branch master into league-team-player-categories

VasilyShelkov committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

Conflicts:
	src/client/search/categories/YearSelector.js

Commit:
Use cloudflare's cdn

Commit:
Category class for normalising button height

Winwardo commented on src/client/search/categories/CategoryFilters.js:
```diff
@@ -1,26 +1,104 @@
+        const currentYear = moment().year() - 1;
+
+        // for each year tab
+        for (let y = currentYear; y >= 2013; y--) {
```
Make 2013 a const explaining WHY it's 2013 - e.g. like EARLIEST_YEAR_AVAILABLE_FROM_FOOTBALL_API or whatever

Winwardo commented on src/client/search/categories/CategoryFilters.js:
```diff
@@ -1,26 +1,104 @@
+    for (let y = currentYear; y >= 2013; y--) {
       const tabContentClassName =
-        year === currentYear ? 'ui bottom attached active tab segment' : 'ui bottom attached tab segment';
+        y === currentYear ? 'ui bottom attached active tab segment' : 'ui bottom attached tab segment';
```
Maybe mention that "active" means "show a loader"?

Winwardo commented on src/client/search/categories/CategoryFilters.js:
```diff
@@ -1,26 +1,104 @@
+              </div>
+            </div>
+          </div>
         </div>
```
So many divs :P

Winwardo commented on src/client/search/categories/CategoryFilters.js:
```diff
@@ -1,26 +1,104 @@
+                    !this.props.football.leagueTeamsByYear[y].isFetching &&
+                      <span className="ui large purple tag label">
+                        Select a team to add players...
+                      </span>
```
I have mixed feelings about this (lines 82 - 89), but can't deny that it's effective

Winwardo commented on src/client/search/categories/LeagueCategory.js:
```diff
@@ -0,0 +1,70 @@
+                {yearsLeagues.length}
+              </div>
+          }
+          Leagues...
```
Was momentarily confused as to why you had a spread operator in text, til I realised it was an ellipsis haha

Winwardo commented on src/client/search/categories/PlayerCategory.js:
```diff
@@ -0,0 +1,56 @@
+                <i className="add green circle icon float right"></i>
+              </div>
+            </div>
+          ))}
```
Could pull this into a <TeamPlayer /> component, your shout

Winwardo commented on src/client/search/categories/TeamCategory.js:
```diff
@@ -0,0 +1,131 @@
+              const inputLength = e.target.value.length;
+
+              // setting very little timeout so that the below happens after
+              // the class names have have changed properly
```
ewwwwwwwwwwwwwwwwwwwwwwwwwwww

VasilyShelkov replied:
Adding issue

VasilyShelkov committed:
Added key props and got minified json for football data

VasilyShelkov committed:
Merged branch league-team-player-categories into league-team-player-categories

Winwardo commented on src/client/search/categories/TeamCategory.js:
```diff
@@ -0,0 +1,131 @@
+                          </div>
+                        </div>
+                      </div>
+                    ))}
```
Again, this would be a lot more readable in its own <LeagueTeam> component - would that mess up the jQuery in any way?

VasilyShelkov committed:
Made see team players buttons fluid so all same size

Winwardo commented on src/client/search/categories/categoryFiltersReducer.js:
```diff
@@ -19,25 +94,18 @@ const footballSeasonReducer = (state = { isFetching: false, seasons: [] }, actio
-  case REMOVE_FOOTBALL_SEASON: {
-    const oldSeasonsByYear = state.seasonsByYear;
-    delete oldSeasonsByYear[action.year];
+  case RECIEVE_YEARS_FOOTBALL_LEAGUES_TEAMS:
```
*RECEIVE not RECIEVE

Winwardo commented on src/client/search/categories/categoryFiltersReducer.spec.js:
```diff
@@ -0,0 +1,199 @@
+import * as actions from './categoryFilterActions';
+
+// copied from ./footballCategoryFiltersReducer
+const footballCategoryFiltersInitialState = {
```
Rather than copying, could we export the default from footballCategoryFiltersReducer and import it here? That way we can never mismatch them

Winwardo commented on src/client/search/categories/categoryFiltersReducer.spec.js:
```diff
@@ -0,0 +1,199 @@
+    const name = 'example football team';
+    const shortName = 'ExFootTeam';
+    const crestUrl = 'http://exFootTeamCrest.com';
+    it('should select and start fetching the requested football team players', () => {
```
Please add a blank line above this, I got confused

Winwardo commented on src/client/search/searchActions.js:
```diff
@@ -1,11 +1,13 @@
   const searchQuery = {
     type: ADD_SEARCH_TERM,
-    id,
+    id: nextSearchTermId++,
```
Are you sure you don't mean `nextSearchTermId + 1`?
`++` mutates the current obejct with a `+= 1`

VasilyShelkov replied:
Nope, it's meant to mutate it

Winwardo commented on src/client/search/searchActions.spec.js:
```diff
@@ -2,52 +2,50 @@ import { should } from 'chai';
     const expectedAction = {
       type: 'ADD_SEARCH_TERM',
-      id,
+      id: searchTermId++,
```
Same here, is ++ definitely what you want?

VasilyShelkov committed:
PR Comment Fixes

- Made Earliest year from football api available as a const (2013)
- Refactored LeagueItem from LeagueCategory
 - Refactored TeamPlayer from PlayerCategory for cleaner code
- Refactored LeagueTeamsList from LeagueCategory and then LeagueTeam from LeagueTeamsList to make the code more readable and split into proper components
- Changed all mispelt recieve -> receive (constants and normal functions)
- export the CategoryFiltersReducerInitialState to use in test so there's no mismatch ever

VasilyShelkov committed:
Merge branch 'master' into league-team-player-categories
